components/ord-service/src/main/java/com/sap/cloud/cmp/ord/service/config/TokenParserConfig.java

```diff
@@ -0,0 +1,31 @@
+package com.sap.cloud.cmp.ord.service.config;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+import com.sap.cloud.cmp.ord.service.repository.SelfRegisteredRuntimeRepository;
+import com.sap.cloud.cmp.ord.service.token.SubscriptionHelper;
+import com.sap.cloud.cmp.ord.service.token.TokenParser;
+
+@Configuration
+public class TokenParserConfig {
+
+    @Value("${subscription.provider_label_key:subscriptionProviderId}")
+    private String selfRegKey;
+
+    @Value("${subscription.region_key:region}")
+    private String regionKey;
+
+    @Value("${subscription.token_prefix:prefix-}")
+    private String tokenPrefix;
+
+    @Autowired
+    private SelfRegisteredRuntimeRepository repo;
```

Comment:
You can add it as an argument to the buildTokenParser and it will be implicitly autowired.
That way we make explicit that the repo is needed as a dependency to the parser.